### PR TITLE
[risk=no] [RW-6064] Api cleanup criteria

### PR DESCRIPTION
**Pull request by SrushtiGangireddy**
Cleaning up criteria mapping and dao calls from controller.

### CODE REVIEW

File: public-api/src/main/java/org/pmiops/workbench/utils/mappers/CriteriaMapper.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import org.pmiops.workbench.utils.mappers.CommonMappers;
+import org.pmiops.workbench.utils.mappers.MapStructConfig;
+
+@Mapper(
+        config = MapStructConfig.class,
+        uses = {CommonMappers.class})
+public interface CriteriaMapper {
+    Criteria dbModelToClient(DbCriteria db);
+
+    default boolean isNonEmpty(String s) {
```
Comment (freemabd) [inline public-api/src/main/java/org/pmiops/workbench/utils/mappers/CriteriaMapper.java]: why is this method needed?

File: public-api/src/main/java/org/pmiops/workbench/cdr/dao/CBCriteriaDao.java [hunk 2]
```diff
@@ -1,23 +1,23 @@
 package org.pmiops.workbench.cdr.dao;
 
-import org.pmiops.workbench.cdr.model.CBCriteria;
+import org.pmiops.workbench.cdr.model.DbCriteria;
 import org.springframework.data.domain.Pageable;
```
Comment (freemabd) [inline public-api/src/main/java/org/pmiops/workbench/cdr/dao/CBCriteriaDao.java]: nit: Pageable is no longer used, can remove import.

File: public-api/src/main/java/org/pmiops/workbench/cdr/dao/CBCriteriaDao.java [hunk 3]
```diff
@@ -1,23 +1,23 @@
...
 import org.springframework.data.jpa.repository.Query;
 import org.springframework.data.repository.CrudRepository;
 import org.springframework.data.repository.query.Param;
 
 import java.util.List;
 
-public interface CBCriteriaDao extends CrudRepository<CBCriteria, Long> {
+public interface CBCriteriaDao extends CrudRepository<DbCriteria, Long> {
```
Comment (freemabd) [inline public-api/src/main/java/org/pmiops/workbench/cdr/dao/CBCriteriaDao.java:10]: Can we write test cases for all methods in this dao? Like https://github.com/all-of-us/workbench/blob/master/api/src/test/java/org/pmiops/workbench/cdr/dao/CBCriteriaDaoTest.java#L30

File: public-api/src/main/java/org/pmiops/workbench/cdr/dao/CBCriteriaDao.java [hunk 4]
```diff
@@ -1,23 +1,23 @@
...
             " from cb_criteria c join concept co on c.concept_id=co.concept_id where c.concept_id = :conceptId and match(c.synonyms) against(:matchWord in boolean mode) and c.domain_id=:domain\n" +
             "union all\n" +
             "select c.id, c.parent_id, c.domain_id, c.is_standard, c.type, c.subtype, c.concept_id, c.code, c.name, c.value, c.est_count, c.is_group, c.is_selectable, c.has_attribute, c.has_hierarchy, c.has_ancestor_data, c.path, c.synonyms, co.can_select\n" +
             " from cb_criteria c join concept co on c.concept_id=co.concept_id where parent_id in (select distinct id from cb_criteria where concept_id=:conceptId and match(synonyms) against(:matchWord in boolean mode) and domain_id=:domain)\n" +
             "order by cast(est_count as unsigned) desc", nativeQuery = true)
-    List<CBCriteria> findParentCounts(@Param("conceptId") String conceptId,@Param("domain") String domain,@Param("matchWord") String matchWord);
+    List<DbCriteria> findParentCounts(@Param("conceptId") String conceptId,@Param("domain") String domain,@Param("matchWord") String matchWord);
```
Comment (freemabd) [inline public-api/src/main/java/org/pmiops/workbench/cdr/dao/CBCriteriaDao.java:18]: If writing test cases for dao calls that use match(), then you will need to make sure you have this class: https://github.com/all-of-us/workbench/blob/master/api/src/main/java/org/pmiops/workbench/cdr/CommonTestDialect.java#L9
Also, need this in application.properties in src/test/resources: https://github.com/all-of-us/workbench/blob/master/api/src/test/resources/application.properties#L6

File: public-api/src/main/java/org/pmiops/workbench/service/CriteriaService.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    private final CBCriteriaDao criteriaDao;
+    private final CriteriaMapper criteriaMapper;
+
+    @Autowired
+    public CriteriaService(CBCriteriaDao criteriaDao, CriteriaMapper criteriaMapper) {
+        this.criteriaDao = criteriaDao;
+        this.criteriaMapper = criteriaMapper;
+    }
+
+    public CriteriaParentResponse getRolledUpCounts(String conceptId, String domainId) {
```
Comment (freemabd) [inline public-api/src/main/java/org/pmiops/workbench/service/CriteriaService.java:28]: This method can be cleaned up and more efficient, but we'll save that for another PR>

File: public-api/src/main/java/org/pmiops/workbench/cdr/dao/CBCriteriaDao.java (hunk 2 above)
Comment (SrushtiGangireddy) [reply public-api/src/main/java/org/pmiops/workbench/cdr/dao/CBCriteriaDao.java]: Done thanks.

File: public-api/src/main/java/org/pmiops/workbench/utils/mappers/CriteriaMapper.java (hunk 1 above)
Comment (SrushtiGangireddy) [reply public-api/src/main/java/org/pmiops/workbench/utils/mappers/CriteriaMapper.java]: Left it in from the reference method I saw online. Removed it.

File: public-api/src/main/java/org/pmiops/workbench/cdr/dao/CBCriteriaDao.java (hunk 3 above)
Comment (SrushtiGangireddy) [reply public-api/src/main/java/org/pmiops/workbench/cdr/dao/CBCriteriaDao.java:10]: I wrote test case for one method in dao. Have to re-write the other method in dao for now we would need can_select in tree to be able to show the concept in source tree based on the flag. Left a TODO in there for later work.

**Review by freemabd — APPROVED** [review]

**Pull request merged**